subxt/src/error/mod.rs

```diff
@@ -112,6 +115,25 @@ impl BlockError {
     }
 }
 
+/// Extrinsic error.
+#[derive(Clone, Debug, Eq, thiserror::Error, PartialEq)]
+#[non_exhaustive]
+pub enum ExtrinsicError {
+    /// Extrinsic type ID cannot be resolved with the provided metadata.
+    #[error("Extrinsic type ID cannot be resolved with the provided metadata. Make sure this is a valid metadata")]
+    MissingType,
+    /// Expected more extrinsic bytes.
+    #[error("Expected more extrinsic bytes")]
+    InsufficientData,
+    /// Unsupported signature.
+    #[error("Unsupported extrinsic version, only version 4 is supported currently")]
+    /// The extrinsic has an unsupported version.
+    UnsupportedVersion(u8),
+    /// Decoding error.
+    #[error("Cannot decode extrinsic: {0}")]
+    DecodingError(codec::Error),
+}
```

Comment:
Given the above, these can then be a part of `BlockError` or something where needed